src/OSPSuite.Infrastructure.Import/Core/DataFormat/MixColumnsDataFormat.cs

```diff
@@ -37,7 +37,8 @@ protected override string ExtractLloq(string description, IUnformattedData data,
       }
       protected override UnitDescription ExtractUnits(string description, IUnformattedData data, List<string> keys, ref double rank)
       {
-         var units = Regex.Match(description, @"\[.+\]").Value;
+         var units = Regex.Match(description, @"\[([^\]\[]*)\]$").Value;
```

Comment:
Should not we have one place with the logic to extra the unit from a text? Here it's the same code as DataFormatHeaderWithUnit. Why?

Reply:
I have created two helper funtions in the base class to fix the code duplication that you are correctly pointing out. I did not move the whole code out, since the two different implementations vary a bit in the way the handle the incrementation of the ranking.